lib/erb_lint/linters/final_newline.rb

```diff
@@ -23,27 +23,38 @@ def offenses(processed_source)
         return offenses if file_content.empty?
 
         match = file_content.match(/(\n+)\z/)
-        ends_with_newline = match.present?
+        final_newline = match&.captures&.first || ""
 
-        if @new_lines_should_be_present && !ends_with_newline
-          offenses << Offense.new(
-            self,
-            processed_source.to_source_range(file_content.size, file_content.size - 1),
-            'Missing a trailing newline at the end of the file.'
-          )
-        elsif !@new_lines_should_be_present && ends_with_newline
+        if @new_lines_should_be_present && final_newline.size != 1
+          if final_newline.empty?
+            offenses << Offense.new(
+              self,
+              processed_source.to_source_range(file_content.size, file_content.size - 1),
+              'Missing a trailing newline at the end of the file.',
+              :insert
+            )
+          else
+            offenses << Offense.new(
```

Comment:
Feels like we are doing the same as the extra new line linter. Should we leave this out?

Reply:
It's related but not the same, I originally tried to fit this functionality into the other linter but it is more at home here. The extra newline linter allows for 2 consecutive newlines anywhere in the file and warns starting at 3 newlines, whereas this linter checks that a single newline is present at the end of the file. I would need to have a special case for the end of the file in the other linter...

Reply:
Sounds good, looks like you reasoned through it!